apps/server/src/routers/api/v1/media.rs

```diff
@@ -1577,14 +1583,123 @@ async fn start_media_analysis(
 	let _ = enforce_session_permissions(&session, &[UserPermission::ManageLibrary])?;
 
 	// Start analysis job
-	ctx.enqueue_job(AnalyzeMediaJob::new(
-		AnalyzeMediaJobVariant::AnalyzeSingleItem(id),
-	))
-	.map_err(|e| {
-		let err = "Failed to enqueue analyze media job";
-		error!(?e, err);
-		APIError::InternalServerError(err.to_string())
-	})?;
+	ctx.enqueue_job(AnalyzeMediaJob::analyze_media_item(id))
+		.map_err(|e| {
+			let err = "Failed to enqueue analyze media job";
+			error!(?e, err);
+			APIError::InternalServerError(err.to_string())
+		})?;
 
 	APIResult::Ok(())
 }
+
+#[utoipa::path(
+	post,
+	path = "/api/v1/media/:id/dimensions",
+	tag = "media",
+	params(
+		("id" = String, Path, description = "The ID of the media to get dimensions for")
+	),
+	responses(
+		(status = 200, description = "Successfully fetched media dimensions"),
+		(status = 401, description = "Unauthorized"),
+		(status = 403, description = "Forbidden"),
+		(status = 404, description = "Media dimensions not available"),
+		(status = 500, description = "Internal server error"),
+	)
+)]
+async fn get_media_dimensions(
+	Path(id): Path<String>,
+	State(ctx): State<AppState>,
+	session: Session,
+) -> APIResult<Json<Vec<PageDimension>>> {
+	// Fetch the media item in question from the database while enforcing permissions
+	let dimensions_entity =
+		fetch_media_page_dimensions_with_permissions(&ctx, &session, id).await?;
+
+	Ok(Json(dimensions_entity.dimensions))
+}
+
+#[utoipa::path(
+	get,
+	path = "/api/v1/media/:id/page/:page/dimensions",
+	tag = "media",
+	params(
+		("id" = String, Path, description = "The ID of the media to get dimensions for"),
+		("page" = i32, Path, description = "The page to get dimensions for")
+	),
+	responses(
+		(status = 200, description = "Successfully fetched media page dimensions"),
+		(status = 401, description = "Unauthorized"),
+		(status = 403, description = "Forbidden"),
+		(status = 404, description = "Media dimensions not available"),
+		(status = 500, description = "Internal server error"),
+	)
+)]
+async fn get_media_page_dimensions(
+	Path((id, page)): Path<(String, i32)>,
+	State(ctx): State<AppState>,
+	session: Session,
+) -> APIResult<Json<PageDimension>> {
+	// Fetch the media item in question from the database while enforcing permissions
+	let dimensions_entity =
+		fetch_media_page_dimensions_with_permissions(&ctx, &session, id).await?;
+
+	// Get the specific page or 404
+	let page_dimension =
+		dimensions_entity
+			.dimensions
+			.get(page as usize)
+			.ok_or(APIError::NotFound(format!(
+				"No page dimensions for page: {}",
+				page
+			)))?;
+
+	Ok(Json(page_dimension.to_owned()))
+}
+
+async fn fetch_media_page_dimensions_with_permissions(
+	ctx: &Arc<Ctx>,
+	session: &Session,
+	id: String,
+) -> APIResult<PageDimensionsEntity> {
+	// First get user permissions/age restrictions
+	let user = enforce_session_permissions(session, &[UserPermission::ManageLibrary])?;
```

Comment:
I don't think we need any elevated permissions to access page dimensions, this should be public for any authenticated user with access to the book IMO. Did you have differing thoughts?

Reply:
No, I think I just don't completely understand the permissions/authorization scheme being used and so I erred on the side of caution. I see that there's no need for library management permissions here so I changed this line to `let user = get_session_user(&session)?;` does that do what we want?

Reply:
I gotcha, yes I think your changes work perfectly. If you ever want an explainer on the permissions/authorization I'm happy to give a TLDR; on Discord